cmd/virt-chroot/tap-device-maker.go

```diff
@@ -65,11 +75,24 @@ func NewCreateTapCommand() *cobra.Command {
 				return fmt.Errorf("could not parse tap device group: %v", err)
 			}
 
-			if err := createTapDevice(tapName, uint(uid), uint(gid), int(queueNumber), int(mtu)); err != nil {
-				return fmt.Errorf("failed to create tap device named %s. Reason: %v", tapName, err)
-			}
-
-			return nil
+			return createTapDevice(tapName, uint(uid), uint(gid), int(queueNumber), int(mtu))
 		},
 	}
 }
+
+func retry(retryAttempts uint, f func() error) (uint, error) {
```

Comment:
I wonder if it is indeed helpful to return the attempt number where it succeeded. What will you do with that ? (other than logging it). 

What can the person looking at the logs do with that info ? That person cares if the tap was created; not that it failed 3 times before succeeding. And they have that info already from the error msg you've computed below.

I won't fight to get it removed from the method signature, really just wondering.

Reply:
I think it is easier to look for it and looks harmless.
We record the failures in the log to see what error it is, here we record at what attempt it happened so ease it up on the person that reads the logs.

It is probably not a must, but seems helpful and harmless.